### DISCUSSION TITLE
MX Linux

### DISCUSSION THREAD

**Discussion by linkman123xyz** [abraunegg/onedrive#1826] [Q&A] [answered]
Hi @abraunegg
Sorry to call on your time. Love the app, but  I am a little lost at this point. 

I had been using Debian 11 but went to MX after Debian created a whole heap of issues when going to sleep. 

- Kernal 5.10.0.11
- Plasma 5.20.5
- Underying base is Debian 11. 
- Onedrive version is v2.4.15-1~np1

Anyway, I set Onedrive up as normally do, followed the instructions you have to ensure it worked, and everything worked. I got a full shared solder sync, and all was good. I then created a config file to allow for Shared folder syncing and placed it in the ~/.config/onedrive directory. But onedrive does not appear to be loading that config file and loads the default options. 

I have installed as per above on another Debian 11 system and this works perfectly with the same config file. 

Now I get the following crash:

```
Adding 1 OneDrive items for processing from /Client/Kahoona Trading Trust/2016/Annual Compliance/Financial Statements
Thread sleeping due to 'HTTP request returned status code 429' - The request has been throttled
Sleeping for 33 seconds
Adding 9 OneDrive items for processing from /Client/Kahoona Trading Trust/2016/Annual Compliance/Workpaper File
Thread sleeping due to 'HTTP request returned status code 429' - The request has been throttled
Sleeping for 85 seconds

ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 429 ()
  Error Reason:     The request has been throttled
  Error Timestamp:  2022-02-02T04:33:46
  API Request ID:   bfca8eee-b0a8-4531-aa6a-74d5ef7678c8
  Calling Function: queryForChildren()

Giving up on sync after three attempts: JSONValue is not an object
std.json.JSONException@std/json.d(308): JSONValue is not an object
----------------
??:? pure @safe bool std.exception.enforce!(std.json.JSONException).enforce!(bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x560cefaf67a2]
??:? inout pure ref @safe inout(std.json.JSONValue) std.json.JSONValue.opIndex(immutable(char)[]) [0x7f7fc5590220]
??:? [0x560cefb29981]
??:? [0x560cefb29d4d]
??:? [0x560cefb117bc]
??:? [0x560cefb00a83]
??:? [0x560cefaff2cf]
??:? [0x560cefb4ae70]
??:? [0x560cefb42604]
??:? void rt.dmain2._d_run_main2(char[][], ulong, extern (C) int function(char[][])*).runAll() [0x7f7fc528407b]
??:? _d_run_main2 [0x7f7fc5283e97]
??:? _d_run_main [0x7f7fc5283ced]
??:? __libc_start_main [0x7f7fc4e8cd09]
??:? [0x560cefad7029]
std.net.curl.CurlException@std/net/curl.d(4402): A libcurl function was given a bad argument on handle null
----------------
??:? pure @safe bool std.exception.enforce!(std.net.curl.CurlException).enforce!(bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x7f7fc5536c7c]
??:? void std.net.curl.HTTP.clearRequestHeaders() [0x7f7fc559dd6f]
??:? [0x560cefb441c3]
??:? void rt.dmain2._d_run_main2(char[][], ulong, extern (C) int function(char[][])*).runAll() [0x7f7fc528407b]
??:? _d_run_main2 [0x7f7fc5283e97]
??:? _d_run_main [0x7f7fc5283ced]
??:? __libc_start_main [0x7f7fc4e8cd09]
??:? `[0x560cefad7029]`
```

So now a little confused. 

Any help would be appreciated. 

BTW my config files is:

```
config file has been updated, checking if --resync needed
onedrive version                       = v2.4.15-1~np1
Config path                            = /home/col/.config/onedrive
Config file found in config path       = false
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /home/col/OneDrive
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = true
business_shared_folders contents:
#Set up Sync for buisness shared. 
PPSR
#Notebooks
Client
second one
cole
```

Any help would be appreciated. Sorry if I have missed an obvious step.

**Comment by abraunegg** [comment]
@linkman123xyz 
> Anyway, I set Onedrive up as normally do, followed the instructions you have to ensure it worked, and everything worked. I got a full shared solder sync, and all was good. I then created a config file to allow for Shared folder syncing and placed it in the ~/.config/onedrive directory. But onedrive does not appear to be loading that config file and loads the default options.

When running on this system - what does `onedrive --version` detail and what does `onedrive --display-config` show?

Are you running via sudo at all? If you are running any of these commands via sudo - for example: `sudo onedrive <options>` - you are effectively running the application as 'root' not your normal user .. so .. dont use sudo. This would be why the client is not loading your configuration file.

> Now I get the following crash ...

Please provide a debug log - use `--verbose --verbose` to generate - so that it can be accurately pinpointed where the failure is occurring. It looks like that a value is not being returned correctly when the 429 response is being handled - but without the debug log it nigh impossible to nail down where. Please provide this ASAP.

**Reply by linkman123xyz** [reply]
Thanks so muchh for the reply. 

> `onedrive --version`
--- onedrive v2.4.15-1~np1

> Are you running via sudo at all?
No I am running as me. 


And the debug log is. 
```
colinl@mx:~$ onedrive --verbose --verbose
[DEBUG] homePath: HOME environment variable set
[DEBUG] homePath: /home/colinl
[DEBUG] configDirBase: WARNING - no XDG_CONFIG_HOME environment variable set
[DEBUG] configDirBase: /home/colinl/.config
[DEBUG] configDirName: Configuring application to use default config path
Using 'user' Config Dir: /home/colinl/.config/onedrive
Using 'system' Config Dir: /etc/onedrive
[DEBUG] refreshTokenFilePath = /home/colinl/.config/onedrive/refresh_token
[DEBUG] deltaLinkFilePath = /home/colinl/.config/onedrive/delta_link
[DEBUG] databaseFilePath = /home/colinl/.config/onedrive/items.sqlite3
[DEBUG] databaseFilePathDryRun = /home/colinl/.config/onedrive/items-dryrun.sqlite3
[DEBUG] uploadStateFilePath = /home/colinl/.config/onedrive/resume_upload
[DEBUG] userConfigFilePath = /home/colinl/.config/onedrive/config
[DEBUG] syncListFilePath = /home/colinl/.config/onedrive/sync_list
[DEBUG] systemConfigFilePath = /etc/onedrive/config
[DEBUG] businessSharedFolderFilePath = /home/colinl/.config/onedrive/business_shared_folders
Configuration file successfully loaded
[DEBUG] sync_dir: Getting syncDir from config value sync_dir
[DEBUG] syncDir: /home/colinl/OneDrive
Using logfile dir: /var/log/onedrive/
[DEBUG] Process ID: 7F10976BE520
[DEBUG] User UID:   1000
[DEBUG] User GID:   1000
[DEBUG] User Name:  colinl
[DEBUG] Testing network to ensure network connectivity to Microsoft OneDrive Service
[DEBUG] Attempting to contact online service
[DEBUG] Shutting down HTTP engine as sucessfully reached OneDrive Online Service
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
[DEBUG] Configured authUrl:          https://login.microsoftonline.com/common/oauth2/v2.0/authorize
[DEBUG] Configured redirectUrl:      https://login.microsoftonline.com/common/oauth2/nativeclient
[DEBUG] Configured tokenUrl:         https://login.microsoftonline.com/common/oauth2/v2.0/token
[DEBUG] Configured driveUrl:         https://graph.microsoft.com/v1.0/me/drive
[DEBUG] Configured driveByIdUrl:     https://graph.microsoft.com/v1.0/drives/
[DEBUG] Configured sharedWithMeUrl:  https://graph.microsoft.com/v1.0/me/drive/sharedWithMe
[DEBUG] Configured itemByIdUrl:      https://graph.microsoft.com/v1.0/me/drive/items/
[DEBUG] Configured itemByPathUrl:    https://graph.microsoft.com/v1.0/me/drive/root:/
[DEBUG] Configured siteSearchUrl:    https://graph.microsoft.com/v1.0/sites?search
[DEBUG] Configured siteDriveUrl:     https://graph.microsoft.com/v1.0/sites/
[DEBUG] Downgrading all HTTP operations to HTTP/1.1 by default
[DEBUG] clientId    = d50ca740-c83f-4d1b-b616-12c519384f0c
[DEBUG] companyName = abraunegg
[DEBUG] appTitle    = OneDrive Client for Linux

--synchronize or --monitor switches missing from your command line input. Please add one (not both) of these switches to your command line or use --help for further assistance.
```

Thanks

**Reply by abraunegg** [reply]
@linkman123xyz 
The version details are needed from your MX platform - the one where the configuration file is not being loaded ....


> And the debug log is.

I need the debug log when the crash when the 429 error is occurring .. not when you are just running the application

**Reply by linkman123xyz** [reply]
> The version details are needed from your MX platform - the one where the configuration file is not being loaded ....
Yes this is the version that I have provided? 
onedrive v2.4.15-1~np1

Just running the debug now but it will take a bit of time as the shared folder is large.

**Reply by linkman123xyz** [reply]
Hey sorry. The log file did not write to the log directory. I have it running now but I am in Australia and it won't be done for a few hours and that will be past midnight. 

I'll upload it once it finishes in the morning. 

Thanks.

**Reply by abraunegg** [reply]
@linkman123xyz 
> I have it running now but I am in Australia

NP ... I am in Melbourne .. rather than posting the log on here - email to the support address and I will pick that up in the morning as well.

**Reply by linkman123xyz** [reply]
Really. I am in Melbourne too. Will do. Thanks for the help.

**Comment by abraunegg** [comment, marked as answer]
@linkman123xyz 
Many thanks for the debug log.

Please can you test the following PR to resolve your issue:

```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/1828/head:pr1828
git checkout pr1828
./configure; make clean; make;
```
To run the PR, you need to run the client from the PR build directory:
```
./onedrive <any options needed>
```

When running the PR, your version should be: `onedrive v2.4.15-14-g657d212` or greater.

**Reply by linkman123xyz** [reply]
Thanks - i'll o this for you shortly.

**Reply by linkman123xyz** [reply]
Thank you sir. 

That is now working as expected. Really appreciate your work on this.

**Reply by linkman123xyz** [reply]
but under moitor i still get the 429 error

**Reply by linkman123xyz** [reply]
All good mate. I d get the error but its is working and syicing correctly. Not sure if that is still an issue moving forward though.

**Reply by abraunegg** [reply]
> All good mate. I d get the error but its is working and syicing correctly. Not sure if that is still an issue moving forward though.

It will be - because you still have the 'package' version installed - not the PR version - meaning you will encounter the same issue. You need to remove the package and install the PR version, then, once v2.4.16 is released, go back to using the package version.

**Comment by abraunegg** [comment]
@linkman123xyz 
> That is now working as expected. Really appreciate your work on this.

Glad the PR resolved your issue

> but under moitor i still get the 429 error

This is because you are still running the `v2.4.15-1~np1` package version.

What you need to do:
1. Stop any 'systemd' service
2. Uninstall the 'packaged' version of the client - `v2.4.15-1~np1`
3. From the PR directory install the PR version with `make install` or you may need to use `sudo make install` depending on how you are logged into your system
4. re-enable any 'systemd' service

**Reply by linkman123xyz** [reply]
Ok thats great. I'll jump onto that shortly. Again appreciate the help
